src/App.tsx

```diff
@@ -15,48 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((prev, next) => {
+    switch (sortType) {
+      case SortType.NONE: 
+        return 0;
+      case SortType.LENGTH:
+        return prev.length - next.length;
+
+      case SortType.ALPHABET:
+        return prev.localeCompare(next);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const goods = useMemo(() => {
+    return getReorderedGoods(goodsFromServer, { sortType, isReversed });
+  }, [goodsFromServer, isReversed, sortType]);
+
+  const reverse = () => setIsReversed(!isReversed);
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortType(SortType.ALPHABET)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {'is-light': sortType !== SortType.ALPHABET})}
```

Comment:
```suggestion
          className={cn('button is-info', {
            'is-light': sortType !== SortType.ALPHABET,
          })}
```
![Screenshot from 2023-07-22 15-53-40](https://github.com/mate-academy/react_list-of-goods-hooks/assets/33029864/82c748b2-6365-4704-a341-7cfdc8519670)